### PR TITLE
Add pre commit hooks

### PR DESCRIPTION
**Pull request by dnephin**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+- repo: git://github.com/pre-commit/pre-commit-hooks
+  sha: 5fe82b
+  hooks:
+    - id: autopep8-wrapper
+      args: ['-i', '--ignore=E309']
```
Comment (analogue) [inline .pre-commit-config.yaml]: I kinda like E309 - Add missing blank line (after class declaration).

Comment (dnephin) [inline .pre-commit-config.yaml]: I think this one doesn't work correctly with docstrings. A blank line makes sense when there is no docstring (just a function), but this was adding blank lines between the class declaration and the docstring, which looked very wrong.

**Comment by analogue** [comment]
ship-it!

**Comment by dnephin** [comment]
rebased on master

**Pull request merged**